### PR TITLE
Xrddev 306

### PR DESCRIPTION
**Pull request by Riippi**
Included:
- Filtering clients list
- Sorting clients list
- The clients are presented in a list that is emulating a tree structure. This is achieved by adding artificial "sorting names" for all items and using them in sorting. Also data table's filter and sort methods are customised.
- When sorting by status or filter is used the "tree view" is made flat so it's not confusing for the user

Related issue: https://jira.niis.org/browse/XRDDEV-306

### CODE REVIEW

File: src/proxy-ui-api/frontend/src/views/Diagnostics.vue [hunk 1]
```diff
@@ -1,25 +1,351 @@
...
+          return 'status-red-ring';
+        case 'global error':
+          return 'status-red';
+        default:
+          return '';
+      }
+    },
+
+    openClient(item: object): void {
+      console.log('edit');
```
Comment (carohauta) [inline src/proxy-ui-api/frontend/src/views/Diagnostics.vue]: Is the `console.log` removed in the end by the minifier?

File: src/proxy-ui-api/frontend/src/views/Diagnostics.vue [hunk 2]
```diff
@@ -1,25 +1,351 @@
...
+  max-width: 300px;
+}
+
+.data-table {
+  width: 100%;
+}
+
+.full-width {
+  width: 100%;
+  max-width: 1280px;
```
Comment (carohauta) [inline src/proxy-ui-api/frontend/src/views/Diagnostics.vue]: Maybe SASS variables could be utilized with certain widths

Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/Diagnostics.vue]: Yes, there will be some common value or limit that will control the width of all views.

File: src/proxy-ui-api/frontend/src/views/Diagnostics.vue (hunk 1 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/Diagnostics.vue]: I removed that

**Pull request merged**